docs/articles/documentation/extending-testcafe/custom-reporter/creating-reporter-plugin/index.md

```diff
@@ -0,0 +1,197 @@
+
+You have to implement four methods which will be further mixed in into the reporter.
+
+```js
+reportTaskStart (/* startTime, userAgents, testCount */) {
+    throw new Error('Not implemented');
+},
+
+reportFixtureStart (/* name, path */) {
+    throw new Error('Not implemented');
+},
+
+reportTestDone (/* name, errs, durationMs, unstable, screenshotPath */) {
+    throw new Error('Not implemented');
+},
+
+reportTaskDone (/* endTime, passed */) {
+    throw new Error('Not implemented');
+}
+```
+
+These methods should output the desired information at certain moments during test run.
+
+* **reportTaskStart** - fires when a test task is started, that is at the beginning of testing.
+
+* **reportFixtureStart** - fires each time a fixture is started.
+
+* **reportTestDone** - fires each time a test is finished.
+
+* **reportTaskDone** - fires when the entire task is finished, that is at the end of testing.
+
+All the required data is provided to these methods through their parameters.
+
+To output information, use [helper methods](/testcafe/documentation/extending-testcafe/custom-reporter-plugin/helper-methods/) and libraries exposed by the object. To refer to the helper methods, use `this`.
+
+**Example**
+
+The following example demonstrates how you can implement four main methods,
+so that a generated report contains information about user agents used for testing,
+a result of individual test execution, summary information about passed/failed tests,
+and overall duration of tests.
+
+```js
+reportTaskStart (startTime, userAgents, testCount) {
+    this.startTime = startTime;
+    this.testCount = testCount;
+
+    this.write(`Running tests in: ${userAgents}`)
+        .newline()
+        .newline();
+},
+
+reportFixtureStart (name) {
+    this.currentFixtureName = name;
+},
+
+reportTestDone (name, errs) {
+    var hasErr    = !!errs.length;
+    var result    = hasErr ? `passed` : `failed`;
+
+    name = `${this.currentFixtureName} - ${name}`;
+
+    var title = `${result} ${name}`;
+
+    this.write(title)
+        .newline();
+},
+
+reportTaskDone (endTime, passed) {
+    var durationMs  = endTime - this.startTime;
+    var durationStr = this.moment.duration(durationMs).format('h[h] mm[m] ss[s]');
+    var footer      = passed === this.testCount ?
+                      `${this.testCount} passed` :
+                      `${this.testCount - passed}/${this.testCount} failed`;
+
+    footer += ` (Duration: ${durationStr})`;
+
+    this.newline()
+        .write(footer)
+        .newline();
+}
+```
+
+## Building Reporter
+
+You can build a reporter project by using the `build` Gulp task.
+
+```bash
+gulp build
+```
+
+## Testing Reporter
+
+To test your reporter works well, you can use the `test` Gulp task.
+To get this task to work, you need to provide baseline reports in your format.
+During testing, reports generated by the custom reporter will be compared with these baseline reports.
```

Comment:
I still don't like `baseline` term. Really haven't seen it in this context elsewhere. Isn't `reference` more suitable word?